helm/external-dns-app/templates/_helpers.tpl

```diff
@@ -22,5 +22,48 @@ helm.sh/chart: {{ include "chart" . | quote }}
 Selector labels
 */}}
 {{- define "labels.selector" -}}
-app: {{ .Values.name | quote }}
+app: {{ .Release.Name | quote }}
 {{- end -}}
+
+{{/*
+Create the list of domains to update
+*/}}
+{{- define "domain.list" -}}
+{{- if .Values.externalDNS.domainFilterList -}}
+{{- range .Values.externalDNS.domainFilterList }}
+{{ printf "- --domain-filter=%s" . }}
+{{- end }}
+{{- else }}
+{{- if eq .Values.aws.access "external" }}
+{{- printf "- --domain-filter=%s" .Values.aws.baseDomain }}
+{{- else }}
+{{- printf "- --domain-filter=%s" .Values.baseDomain }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{/*
+Set the zone type when running on AWS
+*/}}
+{{- define "zone.type" -}}
+{{- if eq .Values.aws.access "external" }}
+{{ printf "- --aws-zone-type=%s" "public" }}
+{{- else }}
+{{- if .Values.aws.zoneType }}
+{{ printf "- --aws-zone-type=%s" .Values.aws.zoneType }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{/*
+Set the role name for KIAM
+*/}}
+{{- define "aws.iam.role" -}}
+{{- if .Values.aws.iam.customRoleName }}
+{{- printf "%s" .Values.aws.iam.customRoleName }}
+{{- else }}
+{{- if eq .Values.aws.access "internal" }}
+{{- printf "%s-Route53Manager-Role" .Values.clusterID }}
+{{- end }}
+{{- end }}
+{{- end }}
```

Comment:
use the custom role name if the user provides one, if not default to our pre-created IAM role.